lib/internal.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Operation = require("./operation");
+const requester = require("./requester.js");
+const mlutil = require("./mlutil");
+let requestOptions;
+let clientObject;
+
+    function intialize(client){
+        clientObject = client;
+        requestOptions = mlutil.copyProperties(clientObject.getConnectionParams());
+        return this;
+    }
+
+    function sendRequest(userOptions, optionalName,  requestType='single', responseType='single'){
+        if(!userOptions.path){
+            throw new Error('Path is needed to send request.');
+        }
+        mlutil.copyProperties(userOptions,requestOptions);
+        let operation = new Operation(
+            optionalName?optionalName:'', clientObject, requestOptions, requestType, responseType
+        );
+        operation.validStatusCodes = [200, 201, 204, 404];
```

Comment:
This looks like it needs to be another arg to `sendRequest` with a default value of `[200, 201, 204, 404]`. Those may not be the correct codes for any given endpoint, so the user needs to be able to override them. 

So perhaps:

    sendRequest("/v1/ping", myOptions, null, "single", "single", [200, 204]);

Reply:
But what if the user inputs 400 or 500 as validStatusCodes?

Reply:
That's a user decision and it's up to the user to pick sensible status codes. The more likely use case is the user only wants to specify a single status code as a valid response, such as `[200]`.